File: Java/day33.java
import java.util.Scanner;

public class PalindromeWithoutBuiltIn {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a string: ");
        String inputString = scanner.nextLine();

        boolean isPalindrome = true;
        int n = inputString.length();
        for (int i = 0; i < n / 2; i++) {
            if (inputString.charAt(i) != inputString.charAt(n - i - 1)) {
                isPalindrome = false;
                break;
            }
        }

        if (isPalindrome) {
            System.out.println("\"" + inputString + "\" is a palindrome!");
        } else {
            System.out.println("\"" + inputString + "\" is not a palindrome.");
        }

        scanner.close();
    }
}

File: Python/day33.py
def is_palindrome(s):
    n = len(s)
    for i in range(n // 2):
        if s[i] != s[n - i - 1]:
            return False
    return True

strrringg= input("Enter a string: ")
if is_palindrome(strrringg):
    print(f'"{strrringg}" is a palindrome!')
else:
    print(f'"{strrringg}" is not a palindrome.')
